File: CS/ExpressionEvaluator Filter/DataViewModel.cs
using DevExpress.Mvvm;
using System.Collections.ObjectModel;

namespace ExpressionEvaluator_Filter {
    class DataViewModel : ViewModelBase {
        public DataViewModel() {
            CreateData(10);
        }

        private void CreateData(int length) {
            Data = new ObservableCollection<DataModel>();
            for(int i = 0; i < length; i++) {
                Data.Add(new DataModel { Text = "Row" + i, Number = i });
            }
        }

        public ObservableCollection<DataModel> Data {
            get { return GetProperty(() => Data); }
            set { SetProperty(() => Data, value); }
        }

        public ObservableCollection<object> VisibleData {
            get { return GetProperty(() => VisibleData); }
            set { SetProperty(() => VisibleData, value); }
        }
    }
}

File: VB/ExpressionEvaluator Filter/DataViewModel.vb
Imports DevExpress.Mvvm
Imports System.Collections.ObjectModel

Namespace ExpressionEvaluator_Filter

    Friend Class DataViewModel
        Inherits ViewModelBase

        Public Sub New()
            CreateData(10)
        End Sub

        Private Sub CreateData(ByVal length As Integer)
            Data = New ObservableCollection(Of DataModel)()
            For i As Integer = 0 To length - 1
                Data.Add(New DataModel With {.Text = "Row" & i, .Number = i})
            Next
        End Sub

        Public Property Data As ObservableCollection(Of DataModel)
            Get
                Return GetProperty(Function() Me.Data)
            End Get

            Set(ByVal value As ObservableCollection(Of DataModel))
                SetProperty(Function() Data, value)
            End Set
        End Property

        Public Property VisibleData As ObservableCollection(Of Object)
            Get
                Return GetProperty(Function() Me.VisibleData)
            End Get

            Set(ByVal value As ObservableCollection(Of Object))
                SetProperty(Function() VisibleData, value)
            End Set
        End Property
    End Class
End Namespace
